integration-azure-oauth-test/integration-azure-oauth-test-docker/src/test/it/com/xwiki/entraid/test/ui/EntraIdIT.java

```diff
@@ -0,0 +1,167 @@
+        // have given that Admin user the admin right directly but the solution we chose is closer to the XS
+        // distribution.
+        setup.setGlobalRights("XWiki.XWikiAdminGroup", "", "admin", true);
+        setup.createAdminUser();
+        setup.loginAsAdmin();
+
+        // Modify the tenant ID to update the OIDC configuration so that the Microsoft login page is displayed.
+        setup.updateObject(ENTRAID_CONFIGURATION_REFERENCE, ENTRAID_CONFIGURATION_CLASSNAME, 0, "tenantId",
+            "test_tenantId");
+
+        // Configure the OIDC user class for the second user tobe able to test the switch to XWiki user functionality.
+        Object userObject =
+            setup.rest().object(new LocalDocumentReference("XWiki", SECOND_USER_NAME), "XWiki.OIDC.UserClass");
+        userObject.withProperties(TestUtils.RestTestUtils.property("issuer", "some_issuer"));
+        setup.rest().add(userObject);
+        setup.forceGuestUser();
+    }
+
+    @Order(1)
+    @Test
+    void authenticate(TestUtils testUtils)
+    {
+        // Switch the authentication service to OIDC.
+        EntraIDViewPage entraIDViewPage = new EntraIDViewPage();
+        testUtils.loginAsAdmin();
+        entraIDViewPage.goToHomePage();
+        AuthServiceViewPage authServiceViewPage = new AuthServiceViewPage();
+        authServiceViewPage.navigateToAuthenticationAdmin();
+        authServiceViewPage.switchToOIDCAuthenticationService();
+        assertTrue(authServiceViewPage.isOIDCSelected());
+        testUtils.forceGuestUser();
+
+        // Check that the login functionality correctly redirect the guest to Microsoft login page, respectively that
+        // the OIDC bypass login is working.
+        entraIDViewPage.goToHomePage();
+        entraIDViewPage.clickLogin();
+        testUtils.getDriver().waitUntilElementIsVisible(By.id("exceptionMessageContainer"));
+        WebElement loginPageMessage = entraIDViewPage.getMicrosoftContainer();
+        assertTrue(loginPageMessage.getText().contains(
+            "Specified tenant identifier 'test_tenantid' is neither a valid DNS name, nor a valid external domain."));
+        entraIDViewPage.goToHomePage();
+        entraIDViewPage.clickLoginBypass();
+        XWikiLoginViewPage xwikiLoginViewPage = new XWikiLoginViewPage();
+        assertTrue(xwikiLoginViewPage.getXwikiLoginContainer().isDisplayed());
+
+        // Disable the OIDC login bypass and check that is no longer displayed.
+        entraIDViewPage.goToHomePage();
+        testUtils.loginAsAdmin();
+        testUtils.updateObject(ENTRAID_CONFIGURATION_REFERENCE, ENTRAID_CONFIGURATION_CLASSNAME, 0,
+            "enableXWikiLoginGlobal", 0);
+        testUtils.forceGuestUser();
+        entraIDViewPage.goToHomePage();
+        assertFalse(entraIDViewPage.canBypassLogin());
+    }
+
+    @Order(2)
+    @Test
+    void SwitchUserTest(TestUtils testUtils)
```

Comment:
`switchUserToXWikiLogin`
It's not a best practice to mention `test` in the test name